ui/app/AppLayouts/Wallet/stores/RootStore.qml

```diff
@@ -105,9 +96,35 @@ QtObject {
         return store
     }
 
+    // "walletSection" is a context property slow to lookup, so we cache it here
+    readonly property var walletSectionInst: walletSection
+    property var totalCurrencyBalance: walletSectionInst.totalCurrencyBalance
```

Comment:
readonly? 😄 Sorry, should have mentioned all properties from the start.

Reply:
In principle, this property has setter and notifier on nim side, so it could be changed both in nim and QML. There is a task: [14802](https://github.com/status-im/status-desktop/issues/14802) to go through all backend properties and adjust them.
However, after checking `totalCurrencyBalance` in QML is always rvalue, so it safely could be readonly.

Reply:
> In principle, this property has setter and notifier on nim side, so it could be changed both in nim and QML. There is a task: [14802](https://github.com/status-im/status-desktop/issues/14802) to go through all backend properties and adjust them. However, after checking `totalCurrencyBalance` in QML is always rvalue, so it safely could be readonly.

Cool, thanks!

Just to clarify my reasoning behind the `readonly` flag:

If somewhere in the code somebody will do this the nim QProperty will not be updated. Only the RootStore. totalCurrencyBalance will have another binding. And this behaviour is not correct.
```
onClick: {
        RootStore.totalCurrencyBalance = otherBalance
}
```

Would be nice to avoid this situation.
So the complete approach IMO would be to have all store properties pointing to nim QProperties as readonly no matter how nim defined the property. And the only way to update a nim QProperty to be through store setter functions.

In this case the correct set function would be something like this:
```
RootStore.qml
    readonly property var totalCurrencyBalance: walletSectionInst.totalCurrencyBalance
    function setCurrencyBalance(otherBalance) {
           walletSectionInst.totalCurrencyBalance = otherBalance
    }

```